### ISSUE TITLE
non-corect normal in create_cube()

### ISSUE THREAD

**Issue by VlasenkoDmytro** [vispy/vispy#1319]
In "generation.py" 

>     # Face Normals 
>     n = np.array([[0, 0, 1], [1, 0, 0], [0, 1, 0],
>                   [-1, 0, 1], [0, -1, 0], [0, 0, -1]])

All normals parallel to axes.
[-1, 0, 1] is not parallel to any axes.
Then correct write:
```
    # Face Normals (All normals parallel to axes)
    n = np.array([[0, 0, 1], [1, 0, 0], [0, 1, 0],
                  [-1, 0, 0], [0, -1, 0], [0, 0, -1]])
```

Such issue there is in https://github.com/vispy/vispy/blob/master/examples/basics/gloo/rotate_cube.py

 ```
   # Face Normals
    n = [[0, 0, 1], [1, 0, 0], [0, 1, 0],
        [-1, 0, 1], [0, -1, 0], [0, 0, -1]]
```

**Comment by VlasenkoDmytro** [comment]
Such issue there is in [https://github.com/vispy/vispy/blob/master/examples/basics/gloo/rotate_cube.py](url)

 ```
   # Face Normals
    n = [[0, 0, 1], [1, 0, 0], [0, 1, 0],
        [-1, 0, 1], [0, -1, 0], [0, 0, -1]]
```

**Comment by davidh-ssec** [comment]
@VlasenkoDmytro I haven't looked in to how exactly the normal is used in the GLSL and the final cube that is generated, but are you seeing something wrong in the output of `rotate_cube.py`? When I run this example as is, it looks like a normal rotating cube and I'm not sure I see anything wrong.

**Comment by davidh-ssec** [comment]
@VlasenkoDmytro Ok I actually looked at it and you seem to be right. Would you mind fixing the code and creating a pull request?

**Comment by djhoese** [comment]
Closed by #2027
